lib/options.nix

```diff
@@ -427,21 +427,28 @@ rec {
      Placeholders will not be quoted as they are not actual values:
        (showOption ["foo" "*" "bar"]) == "foo.*.bar"
        (showOption ["foo" "<name>" "bar"]) == "foo.<name>.bar"
+       (showOption ["foo" "<myPlaceholder>" "bar"]) == "foo.<myPlaceholder>.bar"
   */
   showOption = parts: let
+    # If the part is a named placeholder of the form "<...>" don't escape it.
+    # Required for compatibility with: namedAttrsOf
+    # Can lead to misleading escaping if somebody uses literally "<...>" in their option names.
+    # This is the trade-off to allow for named placeholders in option names.
```

Comment:
Right, we actually have two types of option paths: abstract ones containing placeholders, and real ones that are just data.
Treating them the same isn't quite correct, and merging a workaround could complicate a real fix.
Also note that some of these path items are not module options but attrsOf attributes, and `"*"` will occur as an attribute name in certain configurations to represent a "pattern" that matches everything; for example when an HTTP server doesn't responds for an unknown virtual host, etc.

We could solve this in at least two ways

a. Leave the "option path" type as is, but use two functions
  - leave `showOption` as is, suitable for concrete option paths
  - add `showAbstractOption`, which implements these new rules
b. Only improve the representation
  - represent abstract path items with a value like `{ _type = "optionPathPlaceholder"; metaVariable = "name"; __toString = this: "<${this.metaVariable}>";  }`
  
I think (b) may have good backward compatibility and it solves the problem for module options; not just attrsOf keys.
`__toString` is mostly for compatibility with existing code that kind of works when a placeholder _string_ is passed. This happens when generating option docs.

Reply:
This is pre-existing tech debt, so while this is a good opportunity to fix it before making it slightly worse, perhaps this should be done in a follow-up PR instead.

Reply:
I'd prefer to do this in a follow up PR if you are okay with it.